TddShop/TddShop.Cli/Shipment/AncientRomeShippingService.cs

```diff
@@ -20,7 +48,122 @@ public AncientRomeShippingService(IDeliveryService deliveryService)
         /// <param name="order"></param>
         public void ShipOrder(OrderModel order)
         {
-            
+            try
+            {
+                int referencesNumber = _deliveryService.GenerateShipmentReferenceNumber(order.Items.Sum(p => p.Quantity));
+
+                if (referencesNumber == 0)
+                {
+                    throw new ArgumentOutOfRangeException("Unlucky! Romas did not expect a zero!");
+                }
+
+                if (referencesNumber > maxReferenceNumberValue)
+                {
+                    throw new ArgumentOutOfRangeException("Too high number (max 1000, no idea why :D)");
+                }
+
+                if (PairOfBasicRomanSymbolsAndDigitalNumbers.ContainsValue(referencesNumber))
+                    _romanianNumber
+                        .Append(PairOfBasicRomanSymbolsAndDigitalNumbers
+                        .FirstOrDefault(r => r.Value == referencesNumber)
+                        .Key);
+                else
+                {
+                    int[] tabOfNumbers = Array.ConvertAll(referencesNumber.ToString().ToArray(), x => (int)x - 48);
```

Comment:
Takie 48 warto wydzielić do stałej, żeby było wiadomo co robi ten kod. Chwilę się zastanawialiśmy ale nadal nie wiemy o co chodzi :)

Reply:
Po konwertowaniu char na int otrzymujemy kod ASCII jako int, odejmując 48 (czyli zero) otrzymamy poprawną cyfrę.

Reply:
@KrzJurczyk dzięki za odpowiedź. Tak, ma to sens. Ale niestety nie jest zbyt czytelne.
PS Pamiętaj, że nasze review dotyczy zarówno działania jak i czytelności kodu

Reply:
@KrzJurczyk można by użyć int.Parse